internal/auth0/quickstart.go

```diff
@@ -1,10 +1,150 @@
 package auth0
 
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+
+	"github.com/auth0/go-auth0/management"
+	"github.com/mholt/archiver/v3"
+)
+
+const (
+	quickstartsMetaURL            = "https://auth0.com/docs/meta/quickstarts"
+	quickstartsOrg                = "auth0-samples"
+	quickstartsDefaultCallbackURL = "https://YOUR_APP/callback"
+)
+
+type Quickstarts []Quickstart
+
 type Quickstart struct {
-	Name    string   `json:"name"`
-	Path    string   `json:"path"`
-	Samples []string `json:"samples"`
-	Org     string   `json:"org"`
-	Repo    string   `json:"repo"`
-	Branch  string   `json:"branch,omitempty"`
+	Name                 string `json:"name"`
+	AppType              string `json:"appType"`
+	URL                  string `json:"url"`
+	Logo                 string `json:"logo"`
+	DownloadLink         string `json:"downloadLink"`
+	DownloadInstructions string `json:"downloadInstructions"`
+}
+
+func (q Quickstart) SamplePath(downloadPath string) (string, error) {
+	query, err := url.ParseQuery(q.DownloadLink)
+	if err != nil {
+		return "", err
+	}
+
+	return path.Join(downloadPath, query.Get("path")), nil
+}
+
+func (q Quickstart) Download(ctx context.Context, downloadPath string, client *management.Client) error {
+	quickstartEndpoint := fmt.Sprintf("https://auth0.com%s", q.DownloadLink)
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, quickstartEndpoint, nil)
+	if err != nil {
+		return err
+	}
+
+	params := request.URL.Query()
+	params.Add("org", quickstartsOrg)
+	params.Add("client_id", client.GetClientID())
+
+	// Callback URL, if not set, it will just take the default one.
+	callbackURL := quickstartsDefaultCallbackURL
+	if list := client.GetCallbacks(); len(list) > 0 {
+		callbackURL = list[0]
+	}
+	params.Add("callback_url", callbackURL)
+
+	request.URL.RawQuery = params.Encode()
+	request.Header.Set("Content-Type", "application/json")
+
+	response, err := http.DefaultClient.Do(request)
+	if err != nil {
+		return err
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return fmt.Errorf("expected status %d, got %d", http.StatusOK, response.StatusCode)
+	}
+
+	tmpFile, err := os.CreateTemp("", "auth0-quickstart*.zip")
```

Comment:
Is the `*` in the name intentional?

Reply:
👌🏻 Yes, what that does is to ensure we have a randomly generated string at the end of that pattern in place of the `*`. 

For example: `auth0-quickstart968365240.zip`. 

From the docs:

```
// CreateTemp creates a new temporary file in the directory dir,
// opens the file for reading and writing, and returns the resulting file.
// The filename is generated by taking pattern and adding a random string to the end.
// If pattern includes a "*", the random string replaces the last "*".
```